src/transport/SessionHandle.h

```diff
@@ -17,82 +17,69 @@
 
 #pragma once
 
-#include <app/util/basic-types.h>
-#include <lib/core/NodeId.h>
-#include <lib/core/Optional.h>
+#include <lib/support/Variant.h>
+#include <transport/Session.h>
 #include <transport/UnauthenticatedSessionTable.h>
-#include <transport/raw/PeerAddress.h>
 
 namespace chip {
 
 class SessionManager;
 
+/*
+ * @brief A handle to all types of sessions
+ *
+ * A SessionHandle always referring to a valid session, it will never dangling.
```

Comment:
I don't think this is true given the `mSessionManager.GetSecureSession(session.mLocalSessionId);` bit.  This won't dangle, but it can return null, not a session.

Reply:
I suppose the assertion is that given correct code, we expect the session handle to refer to a session that will have existed _at one time_.  It can still definitely be the case that the session has since been evicted.

All of this makes me a bit nervous.  Correct operation here is predicated on the session ID space not wrapping over dangling session handles.  But you can't actually guarantee this with the session ID allocator because nothing in the system can cleanup outstanding session handles when underlying sessions are evicted.

In short, while hiding the underlying session pointers prevents dereference of an invalid pointer, it does NOT prevent this session ID wrap problem.  I bet I could devise a pathological test case where a session is evicted, 65535 subsequent sessions are opened and then closed, one more session is opened and then a dangling session handle from the first, evicted session becomes an incorrect match to the newest session.

To be clear, this isn't a new problem with this PR.  This PR just makes this more obvious by stripping everything but the session ID out of the session handle.

I don't know how this can be fixed unless the session manager can reach into the system and explicitly invalidate outstanding session handles.  The most obvious design would change the session handles into pointers to actual, allocated memory in the session manager.

But we also need to be able to forcibly evict sessions.  This implies a need to be able to execute a failure callback chain for any session that's being forcibly evicted.  This might be possible with addition of a `CancelExchange(session ID)` interface for the exchange manager.

Complicated, but probably doable...

Reply:
Can we perhaps do something similar to file descriptors?  I think it could be workable for the session manager to forcibly close sessions (as I think it does now), but only for the holders of session handles to release session IDs for reuse by the session ID allocator.